src/app/content/components/Page/mediaModalManager.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { ReactNode, useEffect } from 'react';
+import { createPortal } from 'react-dom';
+import MediaModal from './MediaModal';
+import { HTMLElement, MouseEvent, KeyboardEvent, HTMLButtonElement, HTMLImageElement } from '@openstax/types/lib.dom';
+
+export function createMediaModalManager() {
+  let container: HTMLElement | null = null;
+  let setModalContent: ((content: ReactNode) => void) | null = null;
+
+  const open = (content: ReactNode) => {
+    setModalContent?.(content);
+  };
+
+// tslint:disable-next-line:variable-name
+  const MediaModalPortal = () => {
+    const [isOpen, setIsOpen] = React.useState(false);
+    const [modalContent, setContent] = React.useState<ReactNode>(null);
+
+    useEffect(() => {
+      setModalContent = (content) => {
+        setContent(content);
+        setIsOpen(true);
+      };
+      return () => { setModalContent = null; };
+    }, []);
+    if (typeof document === 'undefined') return null;
+
+    return isOpen ? createPortal(
+      <MediaModal isOpen={isOpen} onClose={() => setIsOpen(false)}>
+        {modalContent}
+      </MediaModal>,
+      document.body
+    ) : null;
```

Comment:
the modal can exist and not be open. you can simplify this to 
```suggestion
    setModalContent = (content) => {
      setContent(content);
      setIsOpen(true);
    };

    if (typeof document === 'undefined') return null;

    return createPortal(
      <MediaModal isOpen={isOpen} onClose={() => setIsOpen(false)}>
        {modalContent}
      </MediaModal>,
      document.body
    );
```

and it'll still work i think

Reply:
That's strange, we are both looking at this file right? (line 79) Maybe it's a cache issue?
 
https://github.com/openstax/rex-web/blob/image-preview/src/app/content/components/Page/mediaModalManager.tsx